dp_app/db.py

```
@@ -1,99 +1,77 @@
 import os
 import sqlite3
 
+from contextlib import contextmanager
 
-def get_full_path_to_db(dbname):
-    running_dir = os.path.dirname(os.path.abspath(__file__))
-    return os.path.join(running_dir, '{}'.format(dbname))
 
+# class realization of db connection cntx mngr
+class DatabaseConnection():
 
-def print_all_tables(path_to_db):
-    conn = sqlite3.connect(path_to_db)
-    cursor = conn.cursor()
-    for table in ('table_users', 'table_books', 'table_authors', 'table_userbooks'):
-        print(table, '-' , cursor.execute('SELECT * FROM {}'.format(table)).fetchall(), '\n')
-    conn.close()
+    def __init__(self, path_to_db):
+        self.path_to_db = path_to_db
 
+    def __enter__(self):
+        if not os.path.exists(self.path_to_db):
+            self.conn = sqlite3.connect(self.path_to_db)
+            create_tables_in_db(self.conn)
+        else:
+            self.conn = sqlite3.connect(self.path_to_db)
+        return self.conn
 
-def add_user(dbname, username):
-    path_to_db = get_full_path_to_db(dbname)
-    if not os.path.exists(path_to_db):
-        print('Database {}.db does not exist, can\'t update'.format(dbname))
-        return
-    conn = sqlite3.connect(path_to_db)
-    cursor = conn.cursor()
-    cursor.execute("INSERT INTO table1 VALUES (NULL, ?)", (username,))
-    conn.commit()
-    print('Added username "{}" into {}.db'.format(username, dbname))
-    # print(cursor.execute('SELECT * FROM table1').fetchall())
+    def __exit__(self, exc_type, exc_val, exc_tb):
+        self.conn.close()
```

Лучше обернуть в try/except: pass, на случай если коннекшена нет. И в ините сделать self.conn = None, принято все переменные класса там объявлять даже если они в начале пустые.